explorer/explorer.go

```diff
@@ -445,5 +449,19 @@ func (e *Explorer) Search(id types.Hash256) (SearchType, error) {
 		return SearchTypeContract, nil
 	}
 
-	return SearchTypeInvalid, errors.New("no such element")
+	v2Contracts, err := e.V2Contracts([]types.FileContractID{types.FileContractID(id)})
```

Comment:
Good opportunity to clean this up by implementing it directly in the database. Will reduce IO since you're not loading data you're going to throw away and probably improve performance.